crates/driver/src/domain/competition/score.rs

```diff
@@ -0,0 +1,47 @@
+use crate::{
+    boundary,
+    domain::{eth, mempools},
+};
+
+impl Score {
+    pub fn new(
+        score_cap: eth::U256,
+        revert_protection: &mempools::RevertProtection,
+        objective_value: eth::U256,
+        gas: eth::Gas,
+        gas_price: eth::GasPrice,
+        success_probability: SuccessProbability,
+    ) -> Result<Self, Error> {
+        let boundary = boundary::Score::new(score_cap, revert_protection);
+        boundary.score(objective_value, gas, gas_price, success_probability)
```

Comment:
Since `boundary` (i.e. the `ScoreCalculator` gets dropped right after anyway why not collapse `.new()` and `.score()` into a single free function call and ditch the boundary strict entirely?

Reply:
I think this is only the question of the right place where to convert to the old score calculator, so no correct answer. I can drop it but then I need to do the same in that single free function. No big benefit in doing either way IMO.

Reply:
I guess that's true. Just seemed to me that it makes sense to shrink the API we expose as early as possible to be able to use that cleaner architecture earlier/in more places.
But I don't feel strongly about it se feel free to leave it as is.